src/mscorlib/src/System/Text/ASCIIEncoding.cs

```diff
@@ -78,12 +80,157 @@ public override unsafe int GetByteCount(char* chars, int count)
             return EncodingForwarder.GetByteCount(this, chars, count);
         }
 
-        public override int GetBytes(String chars, int charIndex, int charCount,
-                                              byte[] bytes, int byteIndex)
+        public unsafe override byte[] GetBytes(String s)
+        {
```

Comment:
In reference to implementing the String only override; or the current implementations having long chains of virtual method calls so inheriting from `ASCIIEncoding` or `UTF8Encoding` only needs to override `GetBytes(char* chars, int charCount, byte* bytes, int byteCount)` to override all the calls; except going via an encoder? (am guessing both)